src/OctoshiftCLI.Tests/Octoshift/Services/GithubClientTests.cs

```diff
@@ -303,6 +305,76 @@ public async Task PostAsync_Returns_String_Response()
         actualContent.Should().Be(EXPECTED_RESPONSE_CONTENT);
     }
 
+    [Fact]
+    public async Task PostAsync_With_StreamContent_Returns_String_Response()
+    {
+        // Arrange
+        var expectedResponse = "Expected response content";
+        var handlerMock = new Mock<HttpMessageHandler>();
+
+        handlerMock
+            .Protected()
+            .Setup<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.IsAny<HttpRequestMessage>(),   // Match any HttpRequestMessage
+                ItExpr.IsAny<CancellationToken>())    // Match any CancellationToken
+            .ReturnsAsync(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(expectedResponse, Encoding.UTF8, "application/json")
+            });
+
+        using var httpClient = new HttpClient(handlerMock.Object);
+        var githubClient = new GithubClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, _dateTimeProvider.Object, PERSONAL_ACCESS_TOKEN);
+
+        var stream = new MemoryStream(Encoding.UTF8.GetBytes("Request body content"));
+        var streamContent = new StreamContent(stream);
+        streamContent.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/json");
+
+        // Act
+        var actualContent = await githubClient.PostAsync("http://example.com", streamContent);
+
+        // Assert
+        actualContent.Should().Be(expectedResponse);
```

Comment:
A couple of things here:
1. We have a helper method for creating the hander mock that can be used here.
2. All `IDisposable` variables should be disposed properly. this is trivial in test methods but still set a good example.
```suggestion
        // Arrange
        var stream = new MemoryStream(new byte[] { 1, 2, 3 });
        using var expectedStreamContent = new StreamContent(stream);
        expectedStreamContent.Headers.ContentType = new("application/octet-stream");

        var handlerMock = MockHttpHandler(req => req.Method == HttpMethod.Post && req.Content == expectedStreamContent);
        using var httpClient = new HttpClient(handlerMock.Object);
        var githubClient = new GithubClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, _dateTimeProvider.Object, PERSONAL_ACCESS_TOKEN);

        // Act
        var actualContent = await githubClient.PostAsync("http://example.com", expectedStreamContent);

        // Assert
        actualContent.Should().Be(EXPECTED_RESPONSE_CONTENT);
```